packages/muze-utils/src/common-utils.js

```diff
@@ -1599,6 +1600,49 @@ const nearestSortingDetails = (dataModel) => {
     return nearestSortDerivation ? nearestSortDerivation.criteria : null;
 };
 
+/**
+ * Returns the sort function based on the type of field
+ * @param {number|string} a first value
+ * @param {number|string} b second value
+ * @param {string} sortOrder Order by which field is to be sorted (asc or desc)
+ * @param {string} subType Field subtype
+ * @return {Function} Sort function
+ */
+const getAppropriateSortingFn = (a, b, sortOrder, subType) => {
+    if (typeof sortOrder === DATA_TYPE.FUNCTION) {
+        return sortOrder(a, b);
+    }
+    if (subType === DimensionSubtype.TEMPORAL) {
+        return sortOrder === SORT_ORDER_ASCENDING ? a - b : b - a;
+    }
+    return sortOrder === SORT_ORDER_ASCENDING ? a.localeCompare(b) : b.localeCompare(a);
+};
+
+/**
+ * Sort field based on it's subtype and sorting order
+ * @param {string} subType Field subtype
+ * @param {string} sortOrder Order by which field is to be sorted (asc or desc)
+ * @param {Array} firstVal First sort parameter
+ * @param {number} secondVal Second sort parameter
+ * @return {Function|null} Sorting function
+*/
+const sortFieldByType = (subType, sortOrder, firstVal, secondVal) => {
+    const sortOrderType = typeof sortOrder;
+
+    if (sortOrderType !== DATA_TYPE.STRING && sortOrderType !== DATA_TYPE.FUNCTION) {
+        return null;
+    }
+    if (sortOrderType === DATA_TYPE.STRING &&
+        sortOrder !== SORT_ORDER_ASCENDING &&
+        sortOrder !== SORT_ORDER_DESCENDING) {
+        return null;
+    }
+    if (subType === DimensionSubtype.TEMPORAL || subType === DimensionSubtype.CATEGORICAL) {
```

Comment:
Enums like `DATA_TYPE` and `DimensionSubtype` are being referenced multiple times. Would make sense to destructure the imports.